app/models/admin.rb

```diff
@@ -10,4 +10,18 @@ class Admin < ApplicationRecord
                     uniqueness: { case_sensitive: false },
                     length: { minimum: 6, maximum: 100 },
                     format: { with: URI::MailTo::EMAIL_REGEXP }
+  validates :password, presence: true,
+                       length: { minimum: 8 },
+                       format: { with: %r{[-!$%^&*()_+|~=`{}\[\]:";'<>?,./\w]{8,}} }
+
+  attr_accessor :current_password
+
+  def update_with_password(params, *options)
```

Comment:
1. `options` parameter is deprecated, remove it

Reply:
2. you lost `password_confirmation` it should be mandatory

Reply:
`password_confirmation` presence is validated in devise gem, when calling `super`
![Screenshot from 2021-12-09 17-45-03](https://user-images.githubusercontent.com/26040954/145428522-e04b9f16-3172-4d36-bf78-ff72c285acee.png)